blog/tidy-collaborative-filtering/index.qmd

```diff
@@ -0,0 +1,216 @@
+
+```{r}
+scheme <- binary_rating_matrix |>
+  evaluationScheme(
+    method = "cross-validation",
+    k      = 5,
+    train  = 0.8,
+    given  = -1
+)
+scheme
+```
+
+Here is the list of recommenders that I consider for the backtest with some intuition:
+
+* Random Items: the benchmark case because it just stupidly chooses random stocks from all possible choices.
+* Popular Items: just recommended the most popular stocks to measured by the number of wikifolios that hold the stock.
+* Association Rules: each wikifolio and its portfolio is considered as a transaction. Association rule mining finds similar portfolios across all traders (if a traders has x, y and z in his/her portfolio, then he/she is X% likely of also including w). 
+* Item-Based Filtering: the algorithm calculates a similarity matrix across stocks. Recommendations are then based on the list of most similar stocks to the ones the wikifolio already has in its portfolio. 
+* User-Based Filtering: the algorithm finds a neighborhood of similar wikifolios for each wikifolio (for this exercise it is set to 100 most similar wikifolios). Recommendations are then based on what the most similar wikifolios have in their portfolio. 
+
+For each algorithm, I base the evaluation on 1, 3, 5, and 10 recommendations. This specification means that each algorithm proposes 1 to 10 recommendations to the test portfolios and the evaluation scheme then checks whether the proposals contain the one withheld stock. Note that the evaluation takes a couple of hours, in particular because the Item-Based and User-Based Filtering approaches are quite time-consuming. 
+
+```{r eval = FALSE}
+algorithms <- list(
+  "Random Items"         = list(name  = "RANDOM",  param = NULL),
+  "Popular Items"        = list(name  = "POPULAR", param = NULL),
+  "Association Rules"    = list(name  = "AR", param = list(supp = 0.01, conf = 0.1)),
+  "Item-Based Filtering" = list(name  = "IBCF", param = list(k = 10)),
+  "User-Based Filtering" = list(name  = "UBCF", param = list(method = "Cosine", nn = 100))
+)
+
+number_of_recommendations <- c(1, 3, 5, 10)
+results <- evaluate(
+  scheme,
+  algorithms,
+  type = "topNList",
+  progress = TRUE,
+  n = number_of_recommendations
+)
+```
+
+```{r eval = FALSE, include = FALSE}
+write_rds(results, "../../data/results.rds")
+```
+
+```{r include = FALSE}
+results <- read_rds("../../data/results.rds")
+```
+
+# Evaluate Recommenders
+
+The output of `evaluate()` already provides the evaluation metrics in a structured way. I can simply average the metrics over the cross-validation folds.
+
+```{r}
+results_tbl <- results |>
+  avg() |>
+  map(as_tibble) |>
+  bind_rows(.id = "model")
+```
+
+Now, for each recommender, we are interested the following numbers:
```

Comment:
**Typo**: "interested in the following number:"

Reply:
plural because a list of numbers follows